sympy/assumptions/handlers/order.py

```diff
@@ -222,8 +231,16 @@ class AskPositiveHandler(CommonHandler):
 
     @staticmethod
     def Expr(expr, assumptions):
+        if expr.is_positive is None:
```

Comment:
@smichr . I tried that and ran the assumption tests. Some of the tests (in tests_query and test_refine) seem to have failed -
assert ask(Q.positive(x)) is None
assert ask(Q.even(k**p), Q.even(k) & Q.integer(p) & Q.positive(p)) is True
assert ask(Q.imaginary(x**y), Q.positive(x) & Q.real(y)) is False
assert ask(Q.negative(-x), Q.positive(x)) is True
assert ask(Q.odd(k**p), Q.even(k) & Q.integer(p) & Q.positive(p)) is False

The problem seems to be that in 
fuzzy_and((expr - a.args[0]).is_positive for a in conjuncts(assumptions)) , 
"is_positive" is being checked for all the assumptions . This might lead to incorrect results. For example, if the assumption is Q.negative, i.e in ask(Q.positive(x-3), Q.negative(x-4)), it will incorrectly return True as x-4 < x-3. Similarly, in the case of ask (Q.positive(x-3), Q.even(x-4)), the assumption Q.even does not give us any information about whether or not x-4 is positive but is still being considered in fuzzy_and((expr - a.args[0]).is_positive.

It makes sense to use the fuzzy_and. Should I use the fuzzy_and but check for specific functions in the assumptions (i.e check for (expr - a.args[0]).is_positive if the assumption function belongs to (Q.positive, Q.negative, Q.nonpositive, Q.nonnegative, Q.zero)) ?